extensions/lifecycle/tasks/LifecycleRequeueRestoreArchiveTask.js

```diff
@@ -0,0 +1,134 @@
+const async = require('async');
+
+const { errors } = require('arsenal');
+const ObjectMD = require('arsenal').models.ObjectMD;
+const BackbeatTask = require('../../../lib/tasks/BackbeatTask');
+
+class LifecycleRequeueRestoreArchiveTask extends BackbeatTask {
```

Comment:
Could we reuse `LifecycleResetTransitionInProgressTask` by renaming it and making it handle both transitions and restores ? as the logic is almost entirely the same and we will also need to duplicate the batch reading logic.

Reply:
Did something like what you are talking about by adding inheritence

Now we sill have the same class but it now inherit from `LifecycleRequeueTask` which handle all the common code betwen restore and archive retry (mostly the batching handling is done is `LifecycleRequeueTask`